### PR TITLE
Add alpine type to purl

### PR DESCRIPTION
**Pull request by noqcks**
apk is now in the purl spec: https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#apk

Requires https://github.com/anchore/packageurl-go/pull/9 to be merged, and then I will update deps in this PR.

### CODE REVIEW

File: syft/pkg/cataloger/apkdb/package.go [hunk 1]
```diff
@@ -42,10 +42,8 @@ func packageURL(m pkg.ApkMetadata, distro *linux.Release) string {
 	}
 
 	return packageurl.NewPackageURL(
-		// note: this is currently a candidate and not technically within spec
-		// see https://github.com/package-url/purl-spec#other-candidate-types-to-define
+		packageurl.TypeAlpine,
 		"alpine",
-		"",
```
Comment (kzantow) [inline syft/pkg/cataloger/apkdb/package.go:48]: Something about this seems off -- this is resulting in PURLs of the form: `apk/alpine/<package>`, should they be of this form or `apk/<package>`?

Comment (noqcks) [reply syft/pkg/cataloger/apkdb/package.go:48]: nope, purl-spec says `pkg:apk/alpine/<package>`, same as for other types like `pkg:deb/ubuntu/<package>`

https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#apk

Comment (kzantow) [reply syft/pkg/cataloger/apkdb/package.go:48]: Oh, excellent 👍

**Comment by noqcks** [comment]
@kzantow this is ready for another review

**Review by kzantow — APPROVED** [review]

**Comment by kzantow** [comment]
LGTM 👍

Thanks your patience getting this contribution to the finish line across multiple repos!

**Pull request merged**